src/app/domain/backup/services/managePermissions.ts

```diff
@@ -0,0 +1,40 @@
+import { Platform, PermissionsAndroid, Permission } from 'react-native'
+
+const PLATFORM_VERSION =
+  typeof Platform.Version === 'number'
+    ? Platform.Version
+    : parseInt(Platform.Version, 10)
+
+const ANDROID_PERMISSION =
+  PLATFORM_VERSION >= 33
+    ? ('android.permission.READ_MEDIA_IMAGES' as Permission) // PermissionsAndroid.PERMISSIONS.READ_MEDIA_IMAGES is only available starting RN 0.70
+    : PermissionsAndroid.PERMISSIONS.READ_EXTERNAL_STORAGE
+
+export const hasMediaPermissions = async (): Promise<boolean> => {
+  if (Platform.OS === 'android') {
+    const hasPermission = await PermissionsAndroid.check(ANDROID_PERMISSION)
+    return hasPermission
+  }
+
+  return true
```

Comment:
Does this mean that iOS has no permission mechanism for media items?

Reply:
Permissions is asked automatically at first backup on iOS. On Android, we need to manage it manually. That's why we have this difference in the code. Anyway I do not manage when the user answers no so I will ask what is the expected behavior when the user say no and modify this part.

And if we want to manage permission manually on iOS we will need to add something like https://github.com/zoontek/react-native-permissions